### PR TITLE
Improve asset paths

### PR DESCRIPTION
**Pull request by jakeboxer**
This improves the asset paths in a couple ways
### Explicit namespacing

Instead of making the following GET requests for assets:

```
/assets/app.css
/assets/app.js
/assets/photo.jpg
```

You'll make these ones:

```
/assets/stylesheets/app.css
/assets/javascripts/app.js
/assets/images/photo.jpg
```
### Vendored assets

You can now throw CSS and JS into `/vendor/assets/stylesheets/` and `/vendor/assets/javascripts/`. You can load individual ones by making requests like `GET /assets/stylesheets/vendored.css`, or you can pick them all up automatically into your app bundle by adding `//= require_tree ../../../vendor/assets/javascripts` to your `app.js` and `//= require_tree ../../../vendor/assets/stylesheets` to your `app.css` (the generator does this for you automatically).

/cc @jessicard @fabianperez

### CODE REVIEW

**Comment by jakeboxer** [comment]
This will fix #11 once merged.

File: spec/fixtures/assets/app/assets/shouldfail/shouldfail.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+div {
+  display: none;
```
Comment (jessicard) [inline spec/fixtures/assets/app/assets/shouldfail/shouldfail.css:2]: you're missing a semicolon

File: spec/fixtures/assets/vendor/assets/javascripts/vendored.js [hunk 2]
```diff
@@ -0,0 +1 @@
+alert("i am vendored")
```
Comment (jessicard) [inline spec/fixtures/assets/vendor/assets/javascripts/vendored.js]: you're missing a semicolon (lol this doesnt matter)

**Comment by jessicard** [comment]
Looks awesome. :thumbsup:

**Comment by fabianperez** [comment]
:+1:

File: spec/fixtures/assets/app/assets/shouldfail/shouldfail.css (hunk 1 above)
Comment (jakeboxer) [inline spec/fixtures/assets/app/assets/shouldfail/shouldfail.css:2]: Where?

Comment (jessicard) [inline spec/fixtures/assets/app/assets/shouldfail/shouldfail.css:2]: wtf i didn't see a ; after none earlier, now its there. whatever

Comment (jakeboxer) [inline spec/fixtures/assets/app/assets/shouldfail/shouldfail.css:2]: [i'm tricky like that u see](https://github.com/jakeboxer/hyperloop/commit/191d6ccb4e9b0814a0ee6e2d13dd8da2acb680aa)

Comment (jessicard) [inline spec/fixtures/assets/app/assets/shouldfail/shouldfail.css:2]: dick

File: spec/fixtures/assets/vendor/assets/javascripts/vendored.js (hunk 2 above)
Comment (jakeboxer) [inline spec/fixtures/assets/vendor/assets/javascripts/vendored.js]: Fixed!

File: spec/fixtures/assets/app/assets/shouldfail/shouldfail.css (hunk 1 above)
Comment (jakeboxer) [inline spec/fixtures/assets/app/assets/shouldfail/shouldfail.css:2]: Testing a quoted email reply thing real quick.

**Pull request merged**